[BugFix] BE crash when ingesting nested array from orc

xiaoyong-z opened the pull request:
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #16365

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

If we create a table with a nested `array<array<int>>`, and ingest data into this table with a nested `array<int>`, then BE will crash.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

chaoyli commented on be/test/exec/test_data/orc_scanner/issue_16365_generate.py:
```diff
@@ -0,0 +1,37 @@
+# Copyright 2021-present StarRocks, Inc. All rights reserved.
```
You can place the orc file in the directory.

wanpengfei-git commented:
@Mergifyio backport branch-2.5

wanpengfei-git commented:
@Mergifyio backport branch-2.4

xiaoyong-z committed:
[BugFix] BE crash when ingesting nested array from orc
If we create a table with a nested array<array<int>>, and ingest data into this table with a nested array<int>, then BE will crash.

Signed-off-by: xyz <a997647204@gmail.com>

xiaoyong-z commented:
run starrocks_be_unittest

trueeyu commented:
Why add the orc file?

wanpengfei-git commented:
run starrocks_admit_test

wanpengfei-git commented:
@Mergifyio backport branch-2.5

wanpengfei-git commented:
@Mergifyio backport branch-2.4
